### PR TITLE
fix: remove rendering quirks

### PR DESCRIPTION
**Pull request by UncleSamtoshi**
- prevent login causing double render flow
- default to cache on main query
- fix bug with navigating to earn screen

### CODE REVIEW

File: app/hooks/use-main-query.ts [hunk 1]
```diff
@@ -13,7 +13,6 @@ const useMainQuery = (): useMainQueryOutput => {
   const { convertCurrencyAmount } = usePriceConversion()
   const { data, previousData, error, loading, refetch } = useQuery(MAIN_QUERY, {
     variables: { hasToken },
-    fetchPolicy: "cache-and-network",
```
Comment (daviroo) [inline app/hooks/use-main-query.ts:16]: How much testing did you do for this change? Last time we changed the fetch policy for main query we ended up with lots of bugs.

**Review by daviroo — CHANGES_REQUESTED** [review]
Just one comment regarding testing for the mainquery change.  Looks good otherwise.

File: app/hooks/use-main-query.ts (hunk 1 above)
Comment (UncleSamtoshi) [reply app/hooks/use-main-query.ts:16]: Not a lot of testing on this one. When was the last time we changed the network policy?

Comment (daviroo) [reply app/hooks/use-main-query.ts:16]: It was when we did the work to migrate to API v2, so quite a while ago.  I remember we had lots of issues with transactions not appearing in the transaction list after a call to send a new transaction for instance.  I guess we could merge and test this on `main` but it would need quite a bit of edge case testing before making it's way to prod IMO.

**Review by daviroo — APPROVED** [review]
Approved but needs tested on `main`
